docs/gen-ai/non-normative/models.ipynb

```diff
@@ -84,6 +84,24 @@
     "    class Config:\n",
     "        extra = \"allow\"\n",
     "\n",
+    "class BlobPart(BaseModel):\n",
+    "    type: Literal[\"blob\"] = Field(description=\"The type of the content captured in this part.\")\n",
+    "    mime_type: Optional[str] = Field(default=None, description=\"The IANA MIME type of the attached data.\")\n",
+    "    filename: Optional[str] = Field(default=None, description=\"The name of the file if provided.\")\n",
+    "    content: bytes = Field(description=\"base64 encoded bytes of the attached data.\")\n",
+    "\n",
+    "    class Config:\n",
+    "        extra = \"allow\"\n",
+    "\n",
+    "class FileDataPart(BaseModel):\n",
+    "    type: Literal[\"file_data\"] = Field(description=\"The type of the content captured in this part.\")\n",
+    "    mime_type: Optional[str] = Field(default=None, description=\"The IANA MIME type of the attached data.\")\n",
+    "    uri: Optional[str] = Field(default=None, description=\"A URI referencing attached data.\")\n",
+    "    file_id: Optional[str] = Field(default=None, description=\"An identifier referencing a file that was pre-uploaded to the provider.\")\n",
```

Comment:
Having a uri vs having a file_id feel like 2 different types of part, unless it's possible to have both.

Reply:
I think it would be one or the other. You can make them mutually exclusive in the JsonSchema but it's hard to represent that in python types